pkg/utils/resources/resources.go

```diff
@@ -93,21 +95,84 @@ func Subtract(lhs, rhs v1.ResourceList) v1.ResourceList {
 	return result
 }
 
-// Ceiling calculates the max between the sum of container resources and max of initContainers
-func Ceiling(pod *v1.Pod) v1.ResourceRequirements {
-	var resources v1.ResourceRequirements
+// inspired from https://github.com/kubernetes/kubernetes/blob/e2afa175e4077d767745246662170acd86affeaf/pkg/api/v1/resource/helpers.go#L96
+// https://kubernetes.io/blog/2023/08/25/native-sidecar-containers/
+// Ceiling calculates the max between the sum of container resources and max of initContainers along with sidecar feature consideration
+func podRequests(pod *v1.Pod) v1.ResourceList {
+	requests := v1.ResourceList{}
+	restartableInitContainerReqs := v1.ResourceList{}
+	maxInitContainerReqs := v1.ResourceList{}
+
+	for _, container := range pod.Spec.Containers {
+
+		MergeInto(requests, MergeResourceLimitsIntoRequests(container))
+	}
+
+	for _, container := range pod.Spec.InitContainers {
+		containerReqs := MergeResourceLimitsIntoRequests(container)
+		if lo.FromPtr(container.RestartPolicy) == v1.ContainerRestartPolicyAlways {
+			MergeInto(requests, containerReqs)
+			MergeInto(restartableInitContainerReqs, containerReqs)
+			container.Resources.Requests = restartableInitContainerReqs
+
+		} else {
+			MergeInto(container.Resources.Requests, restartableInitContainerReqs)
```

Comment:
I think you may want to call `maxInitContainerReqs = MaxResources(maxInitContainerReqs, Merge(containerReqs, restartableInitContainerReqs))`